src/core/metrics.cc

```diff
@@ -376,6 +376,9 @@ const bool metric_disabled = false;
 
 
 histogram& histogram::operator+=(const histogram& c) {
+    if (c.sample_count == 0) {
+        return *this;
+    }
```

Comment:
Looks reasonable, by I'm curious if this patch is somehow related to this series, or is just a random improvement you snuck-in as a bonus.

Reply:
I've comment on that at the header, I've included it because it relates to Prometheus optimization, but I can send it in a different PR to make it clearer.